lib/hydra/derivatives.rb

```diff
@@ -118,7 +118,7 @@ def constantize_processor(processor)
     end
 
     def transform_datastream(file_name, transform_directives, opts={})
```

Comment:
Hi, the problem was that if the opts are empty then the only processor that can be used is the image processor. The processor to use is picked up from the options hash. If the options are empty the default is :image.

```
 def initialize_processor(file_name, transform_directives, opts={})
      processor_class(opts[:processor] || :image).new(self, file_name, transform_directives, opts)
end
```

If this spec, spec/units/transcoding_spec.rb, was changed to use transform_datastream instead of transform_file it would likely fail on anything other than the image processing tests.